test/mocks/MockSvmSpokePoolClient.ts

```diff
@@ -0,0 +1,119 @@
+import winston from "winston";
+import { SvmSpokeClient } from "@across-protocol/contracts";
+import { Address } from "@solana/kit";
+import { DepositWithBlock, RelayerRefundExecution, SortableEvent, SlowFillLeaf, Log } from "../../src/interfaces";
+import { getCurrentTime, bnZero, MakeOptional, EventSearchConfig } from "../../src/utils";
+import { SpokePoolUpdate, SvmSpokePoolClient } from "../../src/clients/SpokePoolClient";
+import { HubPoolClient } from "../../src/clients/HubPoolClient";
+import { EventOverrides } from "../../src/clients/mocks/MockEvents";
+import { AcrossConfigStoreClient } from "../../src/clients/AcrossConfigStoreClient";
+import { MockSolanaEventClient } from "./MockSolanaEventClient";
+import { EventWithData, SvmCpiEventsClient, SVMEventNames, unwrapEventData } from "../../src/arch/svm";
+
+// This class replaces internal SpokePoolClient functionality, enabling
+// the user to bypass on-chain queries and inject events directly.
+export class MockSvmSpokePoolClient extends SvmSpokePoolClient {
+  public mockEventsClient: MockSolanaEventClient;
+  private destinationTokenForChainOverride: Record<number, string> = {};
+
+  constructor(
+    logger: winston.Logger,
+    chainId: number,
+    programId = SvmSpokeClient.SVM_SPOKE_PROGRAM_ADDRESS,
+    deploymentBlock: number = 1,
+    eventSearchConfig: MakeOptional<EventSearchConfig, "to"> = { from: 0, maxLookBack: 0 },
+    opts: { hubPoolClient: HubPoolClient | null } = { hubPoolClient: null }
+  ) {
+    super(
+      logger,
+      opts.hubPoolClient,
+      chainId,
+      BigInt(deploymentBlock),
+      eventSearchConfig,
+      null as unknown as SvmCpiEventsClient,
+      programId,
+      null as unknown as Address
+    );
+    this.mockEventsClient = new MockSolanaEventClient();
+    this.latestHeightSearched = deploymentBlock; // needed?
+  }
+
+  setConfigStoreClient(configStore?: AcrossConfigStoreClient): void {
+    this.configStoreClient = configStore;
+  }
+
+  setDestinationTokenForChain(chainId: number, token: string): void {
+    this.destinationTokenForChainOverride[chainId] = token;
+  }
+
+  getDestinationTokenForDeposit(deposit: DepositWithBlock): string {
+    return this.destinationTokenForChainOverride[deposit.originChainId] ?? super.getDestinationTokenForDeposit(deposit);
+  }
+
+  setLatestBlockNumber(blockNumber: number): void {
+    this.latestHeightSearched = blockNumber;
+  }
+
+  async _update(eventsToQuery: string[]): Promise<SpokePoolUpdate> {
+    const from = this.eventSearchConfig.from ? BigInt(this.eventSearchConfig.from) : undefined;
```

Comment:
ooc, should the `from` slot revert to `this.deploymentBlock` in case of not being specified by the caller?

```suggestion
    const from = this.eventSearchConfig.from ? BigInt(this.eventSearchConfig.from) : undefined;
```

Reply:
Yes that's a good idea. Done [here](https://github.com/across-protocol/sdk/pull/1010/files#diff-687f55ed02679a4c62667e363806b6797a5c7830d76f7ce98138c9816f87e87cR58).